pkg/config/parameters.go

```diff
@@ -491,6 +493,10 @@ type Parameters struct {
 	// Server contains parameters for the xDS server.
 	Server ServerParameters `yaml:"server,omitempty"`
 
+	// GatewayClassController is the identifier used by Contour's gatewayclass
+	// controller to determine if a GatewayClass object should be managed.
+	GatewayClassController string `yaml:"gatewayclass-controller,omitempty"`
```

Comment:
Doesn't the gatewayclass reference give you this value automatically? We've got to have another set of config items now in addition to the gateway to configure?

Reply:
This tells the controller how to construct the `gatewayclass.spec.controller` string used for reconciling gatewayclasses. If a cluster has multiple Contours, the gatewayclass-controller for each needs to differentiate its self. For example:
```
kind: GatewayClass
apiVersion: networking.x-k8s.io/v1alpha1
metadata:
  name: public
spec:
  controller: projectcontour.io/public/contour
---
kind: Gateway
apiVersion: networking.x-k8s.io/v1alpha1
metadata:
  name: contour
  namespace: public
spec:
  gatewayClassName: public
<SNIP>
---
kind: GatewayClass
apiVersion: networking.x-k8s.io/v1alpha1
metadata:
  name: private
spec:
  controller: projectcontour.io/private/contour
---
kind: Gateway
apiVersion: networking.x-k8s.io/v1alpha1
metadata:
  name: contour
  namespace: private
spec:
  gatewayClassName: private
<SNIP>
---

We can remove the API group from the string to remove the verbosity. Note that this is internal config and not exposed to users.

Reply:
> This is internal config and not exposed to users.

It's in the public `config` package and specifies yaml parsers, so from the PR it seems like users are supposed to configure this field. 

Should we just look at the configured gateway.spec.controller and verify the class matches that? If not then skip, otherwise process? Seems like we just need a `gateway.spec.parametersRef` and skip the whole gatewayclass object all together.

Reply:
@stevesloka OK, I'm still trying to connect the dots with the Contour code. I'm trying to generate the class name for Contour based on the namespace that Contour is running, i.e. <CONTOUR_NAMESPACE>/contour or even better <CONTOUR_NAMESPACE>/<CONTOUR_DEPLOYMENT_NAME>.